test/ng/compileSpec.js

```diff
@@ -4033,6 +4033,68 @@ describe('$compile', function() {
     });
 
 
+    it('should bind to multiple directives controllers via object notation (new scope)', function() {
+      var controllerCalled = false;
+      var secondControllerCalled = false;
+      module(function($compileProvider, $controllerProvider) {
+        $controllerProvider.register('myCtrl', function() {
+          expect(this.data).toEqualData({
+            'foo': 'bar',
+            'baz': 'biz'
+          });
+          expect(this.str).toBe('Hello, world!');
+          expect(this.fn()).toBe('called!');
+          controllerCalled = true;
+        });
+        $controllerProvider.register('secondCtrl', function() {
+          expect(this.data).toEqualData({
+            'foo2': 'bar2',
+            'baz2': 'biz2'
+          });
+          expect(this.str).toBe('Hello, second world!');
+          expect(this.fn()).toBe('second called!');
+          secondControllerCalled = true;
+        });
+        $compileProvider.directive('fooDir', valueFn({
+          bindToController: {
+            'data': '=dirData',
+            'str': '@dirStr',
+            'fn': '&dirFn'
+          },
+          scope: true,
+          controller: 'myCtrl as myCtrl'
+        }));
+        $compileProvider.directive('barDir', valueFn({
+          bindToController: {
+            'data': '=barData',
+            'str': '@barStr',
+            'fn': '&barFn'
+          },
+          scope: true,
```

Comment:
Believe it or not, I understand what this issue is about --- The problem is the "scope: true" behaviour being shared by multiple directives on the same element. This was never meant to be supported, there is a WONTFIX'd bug about this, because it's explicitly __not__ supported, it's just that we fail to throw the error sometimes (depending on types of scopes used, and directive priorities). This is never something that should be depended on